src/NetTopologySuite/Algorithm/ElevationModel.cs

```diff
@@ -0,0 +1,132 @@
+﻿using NetTopologySuite.Geometries;
+
+namespace NetTopologySuite.Algorithm
+{
+    /// <summary>
+    /// A base elevation model class.
+    /// </summary>
+    public class ElevationModel
+    {
+        private static ElevationModel _default = new ElevationModel();
+
+        private readonly double _z;
+        private readonly Envelope _extent;
+        private readonly int _srid;
+
+        /// <summary>
+        /// Creates an elevation model that always returns <see cref="Coordinate.NullOrdinate"/> as result.
+        /// </summary>
+        public ElevationModel() : this(Coordinate.NullOrdinate)
+        { }
+
+        /// <summary>
+        /// Creates an elevation model that always returns <paramref name="z"/> as result.
+        /// </summary>
+        /// <param name="z">The result value for <see cref="GetZ(Coordinate)"/> or its overloads.</param>
+        internal ElevationModel(double z)
+        {
+            _z = z;
+        }
+
+        /// <summary>
+        /// Creates an elevation model that always returns <paramref name="z"/> as result.
+        /// </summary>
+        /// <param name="z">The result value for <see cref="GetZ(Coordinate)"/> or its overloads.</param>
+        /// <param name="extent">The extent where this elevation model is valid</param>
+        /// <param name="srid">The spatial reference id</param>
+        public ElevationModel(double z, Envelope extent, int srid)
+        {
+            _z = z;
+            _extent = extent;
+            _srid = srid;
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating the default <see cref="ElevationModel"/>
+        /// </summary>
+        /// <remarks>The value <c>null</c> cannot be assigned to this property, it will be converted to a no-op elevation model.</remarks>
+        public static ElevationModel Default
+        {
+            get => _default;
+            set
+            {
+                if (value == null)
+                    value = new ElevationModel();
+                _default = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets a value indicating the extent where this elevation model is valid.
+        /// </summary>
+        /// <remarks>
+        /// If this value is <c>null</c>, no check for the validity of the
+        /// input arguments for <see cref="GetZ(Coordinate)"/> and its overload is made.
+        /// </remarks>
+        public Envelope Extent { get => _extent; }
+
+        /// <summary>
+        /// Gets a value indicating the spatial reference system this elevation model belongs to.
+        /// </summary>
+        public int SRID { get => _srid; }
```

Comment:
> There are some methods on ElevationModel that are currently not used but might help in looking for an appropriate elevation model. We can ditch/change/append these.

I assume that the quote refers to these two, `Extent` and `SRID`. In my opinion:
1. `Extent` is a good idea.
    - Not all publicly available elevation models cover the full globe, and even for the ones that do, someone might want to only download a subset of its huge data.
    - Being able to advertise that "this `ElevationModel` is only valid within this extent" is valuable.
3. `SRID` does not seem as useful.
    - Even ignoring the fact that (I think?) we never use it, there are plenty of cases where we **couldn't** use it (even if we wanted to), and it would restrict design options in the future if we wanted to make this more of a first-class concept in NTS later and discovered that we needed something more (or less) than a single `int` here.
    - Especially since there are both horizontal and vertical datums, and this may be the only place where we care about whether a SRID refers to a specific horizontal datum ("x/y are interpreted according to this SRS"), a specific vertical datum ("z results shall be interpreted according to this SRS") or both.
    - Besides, subclasses can add this themselves if they want to make use of it. I'd get rid of this one until we have **something** more tangible. We can add it later without breaking anyone.